src/System.Memory/tests/SequenceReader/SkipDelimiter.cs

```diff
@@ -143,120 +222,41 @@ public void TryReadTo_SkipDelimiter_Runs()
             Assert.Equal(Encoding.UTF8.GetBytes("^^"), span.ToArray());
             Assert.True(reader.IsNext((byte)'|'));
             Assert.Equal(2, reader.Consumed);
+            reader.Rewind(reader.Consumed);
+            Assert.True(reader.TryReadTo(out sequence, (byte)'|', (byte)'^', advancePastDelimiter: false));
+            Assert.Equal(Encoding.UTF8.GetBytes("^^"), sequence.ToArray());
+            Assert.True(reader.IsNext((byte)'|'));
+            Assert.Equal(2, reader.Consumed);
 
             // Leading run of 3
             bytes = SequenceFactory.CreateUtf8("^^^|abc");
             reader = new SequenceReader<byte>(bytes);
             Assert.False(reader.TryReadTo(out span, (byte)'|', (byte)'^', advancePastDelimiter: false));
             Assert.True(reader.IsNext((byte)'^'));
             Assert.Equal(0, reader.Consumed);
+            Assert.False(reader.TryReadTo(out sequence, (byte)'|', (byte)'^', advancePastDelimiter: false));
+            Assert.True(reader.IsNext((byte)'^'));
+            Assert.Equal(0, reader.Consumed);
 
             // Trailing run of 3
             bytes = SequenceFactory.CreateUtf8("abc^^^|");
             reader = new SequenceReader<byte>(bytes);
             Assert.False(reader.TryReadTo(out span, (byte)'|', (byte)'^', advancePastDelimiter: false));
             Assert.True(reader.IsNext((byte)'a'));
             Assert.Equal(0, reader.Consumed);
+            Assert.False(reader.TryReadTo(out sequence, (byte)'|', (byte)'^', advancePastDelimiter: false));
+            Assert.True(reader.IsNext((byte)'a'));
+            Assert.Equal(0, reader.Consumed);
 
             // Trailing run of 3, split
             bytes = SequenceFactory.CreateUtf8("abc^^^", "|");
             reader = new SequenceReader<byte>(bytes);
             Assert.False(reader.TryReadTo(out span, (byte)'|', (byte)'^', advancePastDelimiter: false));
             Assert.True(reader.IsNext((byte)'a'));
             Assert.Equal(0, reader.Consumed);
-        }
-
-        [Fact]
-        public void TryReadTo_SkipDelimiter_ReadOnlySequence()
```

Comment:
Why remove this?

Reply:
I interleaved it with the other test so we don't miss added scenarios.